protocols/virtualdefund/virtualdefund.go

```diff
@@ -178,6 +151,40 @@ func calculatePaidToBob(proposedFinalState state.State, getChannel GetChannelByI
 	return big.NewInt(0).Sub(finalBobAmount, initialBobAmount), nil
 }
 
+// ConstructObjectiveFromPayload takes in a message and constructs an objective from it.
+// It accepts the message, myAddress, and a function to to retrieve ledgers from a store.
+func ConstructObjectiveFromPayload(
+	p protocols.ObjectivePayload,
+	preapprove bool,
+	myAddress types.Address,
+	getChannel GetChannelByIdFunction,
+	getTwoPartyConsensusLedger GetTwoPartyConsensusLedgerFunction,
+) (Objective, error) {
+
+	switch p.Type {
+	case InitPayload:
+		return Objective{}, fmt.Errorf("unknown payload type %s", p.Type)
```

Comment:
Do you think it is worth having these switch statements in all of the protocols (now)? To me they do a better job at showing the new freedom than the `InitPayload` case itself (that could be removed)? Don't feel hugely strongly, though...

Reply:
I think that's a fair point, but since #866 is pretty close I'm just going to leave things as-is in this PR.